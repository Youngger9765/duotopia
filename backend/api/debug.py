"""Debug endpoints for system testing"""
import os
from fastapi import APIRouter, UploadFile, File, Form
from fastapi.responses import JSONResponse

router = APIRouter(prefix="/api/debug", tags=["debug"])


@router.get("/system-check")
async def system_check():
    """Check system configuration and environment variables"""

    # Check Azure Speech Service configuration
    azure_config = {
        "speech_key_exists": bool(os.getenv("AZURE_SPEECH_KEY")),
        "speech_region": os.getenv("AZURE_SPEECH_REGION", "Not set"),
        "speech_endpoint": os.getenv("AZURE_SPEECH_ENDPOINT", "Not set"),
    }

    # Check Google Cloud Storage configuration
    gcs_config = {
        "bucket_name": os.getenv("GCS_BUCKET_NAME", "Not set"),
        "credentials_exist": bool(os.getenv("GCS_SERVICE_ACCOUNT_KEY")),
        "project_id": os.getenv("GCP_PROJECT_ID", "Not set"),
    }

    # Check database configuration
    db_config = {
        "database_url_exists": bool(os.getenv("DATABASE_URL")),
        "db_host": os.getenv("DB_HOST", "Not set"),
        "db_name": os.getenv("DB_NAME", "Not set"),
    }

    # Check general configuration
    general_config = {
        "environment": os.getenv("ENVIRONMENT", "development"),
        "debug_mode": os.getenv("DEBUG", "false").lower() == "true",
        "cors_origins": os.getenv("CORS_ORIGINS", "Not set"),
    }

    return JSONResponse(
        content={
            "status": "ok",
            "azure": azure_config,
            "gcs": gcs_config,
            "database": db_config,
            "general": general_config,
            "message": "System check completed",
        },
        status_code=200,
    )


@router.get("/health")
async def debug_health():
    """Simple health check for debug purposes"""
    return {
        "status": "healthy",
        "service": "debug",
        "timestamp": os.popen("date").read().strip(),
    }


@router.post("/test-upload")
async def test_upload_no_auth(
    file: UploadFile = File(...),
    duration_seconds: int = Form(5),
):
    """測試音檔上傳 (無認證版本)"""
    try:
        import uuid
        from datetime import datetime

        # 讀取檔案內容
        content = await file.read()

        # 檢查檔案大小
        max_size = 2 * 1024 * 1024  # 2MB
        if len(content) > max_size:
            return JSONResponse(
                status_code=400,
                content={"status": "error", "message": "File too large"},
            )

        # 檢查檔案類型
        allowed_types = [
            "audio/webm",
            "audio/webm;codecs=opus",
            "audio/mp3",
            "audio/wav",
            "audio/mpeg",
            "audio/wave",
            "application/octet-stream",  # 瀏覽器上傳時的通用類型
        ]
        if file.content_type not in allowed_types:
            return JSONResponse(
                status_code=400,
                content={
                    "status": "error",
                    "message": f"Invalid file type: {file.content_type}",
                },
            )

        # 生成檔名
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
        file_id = str(uuid.uuid4())[:8]
        filename = f"debug_upload_{timestamp}_{file_id}_{file.filename}"

        # 儲存到本地
        local_dir = os.path.join(
            os.path.dirname(os.path.dirname(__file__)), "local_recordings"
        )
        os.makedirs(local_dir, exist_ok=True)

        local_path = os.path.join(local_dir, filename)
        with open(local_path, "wb") as f:
            f.write(content)

        return {
            "status": "success",
            "message": "Upload successful (no auth)",
            "audio_url": f"/local_recordings/{filename}",
            "file_info": {
                "filename": file.filename,
                "content_type": file.content_type,
                "size": len(content),
                "local_path": local_path,
            },
        }

    except Exception as e:
        return JSONResponse(
            status_code=500, content={"status": "error", "message": str(e)}
        )


@router.post("/test-azure-speech")
async def test_azure_speech_no_auth(
    file: UploadFile = File(...),
    reference_text: str = Form("Hello world"),
):
    """測試 Azure Speech Assessment (無認證版本)"""
    try:
        import tempfile

        # 保存臨時檔案
        with tempfile.NamedTemporaryFile(delete=False, suffix=".webm") as temp_file:
            content = await file.read()
            temp_file.write(content)
            temp_file_path = temp_file.name

        try:
            # 這裡可以加入 Azure Speech SDK 測試
            # 現在先返回模擬結果
            return {
                "status": "success",
                "message": "Azure Speech test (no auth)",
                "assessment": {
                    "reference_text": reference_text,
                    "pronunciation_score": 85.5,
                    "accuracy_score": 90.2,
                    "fluency_score": 78.9,
                    "completeness_score": 95.0,
                    "mock": True,
                },
                "file_info": {
                    "filename": file.filename,
                    "content_type": file.content_type,
                    "size": len(content),
                },
            }
        finally:
            # 清理臨時檔案
            if os.path.exists(temp_file_path):
                os.unlink(temp_file_path)

    except Exception as e:
        return JSONResponse(
            status_code=500, content={"status": "error", "message": str(e)}
        )


@router.post("/test-gcs-simple")
async def test_gcs_simple_no_auth(
    file: UploadFile = File(...),
):
    """測試 GCS 上傳 (無認證簡化版本)"""
    try:
        content = await file.read()

        # 模擬 GCS 上傳流程
        result = {
            "status": "success",
            "message": "GCS test completed (no auth)",
            "simulation": {
                "would_upload_to": "duotopia-audio/recordings/",
                "file_size": len(content),
                "content_type": file.content_type,
                "filename": file.filename,
                "mock_url": f"https://storage.googleapis.com/duotopia-audio/recordings/mock_{file.filename}",
            },
        }

        # 在本地開發環境，實際儲存到本地
        environment = os.getenv("ENVIRONMENT", "development")
        if environment == "development":
            try:
                import uuid
                from datetime import datetime

                # 生成唯一檔名
                timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
                file_id = str(uuid.uuid4())[:8]
                local_filename = f"debug_test_{timestamp}_{file_id}_{file.filename}"

                # 確保目錄存在
                local_dir = os.path.join(
                    os.path.dirname(os.path.dirname(__file__)), "local_recordings"
                )
                os.makedirs(local_dir, exist_ok=True)

                # 儲存檔案
                local_path = os.path.join(local_dir, local_filename)
                with open(local_path, "wb") as f:
                    f.write(content)

                result["local_storage"] = {
                    "saved": True,
                    "path": local_path,
                    "url": f"/local_recordings/{local_filename}",
                }

            except Exception as local_error:
                result["local_storage"] = {"saved": False, "error": str(local_error)}

        return result

    except Exception as e:
        return JSONResponse(
            status_code=500, content={"status": "error", "message": str(e)}
        )
