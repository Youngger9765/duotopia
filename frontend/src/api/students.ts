import type { Student } from '@/types'

export const studentsApi = {
  getStudents: async (): Promise<Student[]> => {
    // Mock implementation
    return []
  },

  createStudent: async (student: Omit<Student, 'id'>): Promise<Student> => {
    // Mock implementation
    return { ...student, id: 1 } as Student
  },

  updateStudent: async (id: number, student: Partial<Student>): Promise<Student> => {
    // Mock implementation
    return { ...student, id } as Student
  },

  deleteStudent: async (id: number): Promise<void> => {
    // Mock implementation
  },

  resetPassword: async (id: number) => {
    // Mock implementation
    return {
      message: '密碼已重置',
      default_password: '20100101',
    }
  },

  bulkAssignToClass: async (studentIds: number[], classId: number) => {
    // Mock implementation
    return {
      success: true,
      assigned_count: studentIds.length,
    }
  },
}