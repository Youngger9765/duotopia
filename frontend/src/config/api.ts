// API configuration
// This value is injected at build time from environment variables
export const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000'

// Helper function for API calls
export const apiCall = async (endpoint: string, options?: RequestInit) => {
  const url = `${API_URL}${endpoint}`
  const response = await fetch(url, {
    ...options,
    headers: {
      'Content-Type': 'application/json',
      ...options?.headers,
    },
  })
  
  if (!response.ok) {
    throw new Error(`API call failed: ${response.status}`)
  }
  
  return response.json()
}