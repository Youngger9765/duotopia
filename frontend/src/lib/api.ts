import axios from 'axios'

const API_BASE_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000'

export const api = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
})

// Add auth token to requests
api.interceptors.request.use((config) => {
  const token = localStorage.getItem('token')
  if (token) {
    config.headers.Authorization = `Bearer ${token}`
  }
  return config
})

// Handle auth errors
api.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response?.status === 401) {
      localStorage.removeItem('token')
      window.location.href = '/login'
    }
    return Promise.reject(error)
  }
)

// Auth API
export const authApi = {
  login: (email: string, password: string) => {
    const formData = new URLSearchParams({ username: email, password })
    return api.post('/api/auth/login', formData, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      }
    })
  },
  
  googleLogin: (idToken: string) =>
    api.post('/api/auth/google', { id_token: idToken }),
  
  studentLogin: (email: string, birthDate: string) =>
    api.post('/api/auth/student/login', { email, birth_date: birthDate }),
  
  register: (data: any) =>
    api.post('/api/auth/register', data),
  
  validate: () =>
    api.get('/api/auth/validate'),
}

// Teacher API
export const teacherApi = {
  getDashboard: () =>
    api.get('/api/teachers/dashboard'),
  
  createClassroom: (data: any) =>
    api.post('/api/teachers/classrooms', data),
  
  getClassrooms: () =>
    api.get('/api/teachers/classrooms'),
    
  getClassroom: (classroomId: string) =>
    api.get(`/api/teachers/classrooms/${classroomId}`),
    
  getClassroomStudents: (classroomId: string) =>
    api.get(`/api/teachers/classrooms/${classroomId}/students`),
    
  updateClassroom: (classroomId: string, data: any) =>
    api.put(`/api/teachers/classrooms/${classroomId}`, data),
    
  deleteClassroom: (classroomId: string) =>
    api.delete(`/api/teachers/classrooms/${classroomId}`),
  
  addStudentsToClassroom: (classroomId: string, studentIds: string[]) =>
    api.post(`/api/teachers/classrooms/${classroomId}/students`, { student_ids: studentIds }),
  
  createCourse: (data: any) =>
    api.post('/api/teachers/courses', data),
  
  getCourses: () =>
    api.get('/api/teachers/courses'),
    
  getCourse: (courseId: string) =>
    api.get(`/api/teachers/courses/${courseId}`),
    
  updateCourse: (courseId: string, data: any) =>
    api.put(`/api/teachers/courses/${courseId}`, data),
    
  deleteCourse: (courseId: string) =>
    api.delete(`/api/teachers/courses/${courseId}`),
  
  createLesson: (courseId: string, data: any) =>
    api.post(`/api/teachers/courses/${courseId}/lessons`, data),
    
  getLessons: (courseId: string) =>
    api.get(`/api/teachers/courses/${courseId}/lessons`),
  
  createAssignments: (data: any) =>
    api.post('/api/teachers/assignments', data),
}

// Student API
export const studentApi = {
  register: (data: any) =>
    api.post('/api/students/register', data),
  
  searchTeacher: (email: string) =>
    api.get(`/api/students/teachers/search?email=${email}`),
  
  getTeacherClassrooms: (teacherId: string) =>
    api.get(`/api/students/teachers/${teacherId}/classrooms`),
  
  getClassroomStudents: (classroomId: string) =>
    api.get(`/api/students/classrooms/${classroomId}/students`),
  
  getAssignments: (studentId: string) =>
    api.get(`/api/students/${studentId}/assignments`),
  
  getCourses: (studentId: string) =>
    api.get(`/api/students/${studentId}/courses`),
  
  submitAssignment: (assignmentId: string, data: any) =>
    api.post(`/api/students/assignments/${assignmentId}/submit`, data),
  
  updateParentInfo: (studentId: string, parentEmail: string, parentPhone: string) =>
    api.put(`/api/students/${studentId}/parent-info`, { parent_email: parentEmail, parent_phone: parentPhone }),
}

// Admin API
export const adminApi = {
  // Schools/Institutions Management
  getSchools: () =>
    api.get('/api/admin/schools'),
    
  getSchool: (schoolId: string) =>
    api.get(`/api/admin/schools/${schoolId}`),
    
  createSchool: (data: any) =>
    api.post('/api/admin/schools', data),
    
  updateSchool: (schoolId: string, data: any) =>
    api.put(`/api/admin/schools/${schoolId}`, data),
    
  deleteSchool: (schoolId: string) =>
    api.delete(`/api/admin/schools/${schoolId}`),
  
  // Staff/User Management
  getUsers: (params?: { role?: string; school_id?: string }) =>
    api.get('/api/admin/users', { params }),
    
  getUser: (userId: string) =>
    api.get(`/api/admin/users/${userId}`),
    
  createUser: (data: any) =>
    api.post('/api/admin/users', data),
    
  updateUser: (userId: string, data: any) =>
    api.put(`/api/admin/users/${userId}`, data),
    
  deleteUser: (userId: string) =>
    api.delete(`/api/admin/users/${userId}`),
    
  updateUserStatus: (userId: string, status: string) =>
    api.put(`/api/admin/users/${userId}/status`, { status }),
  
  // Student Management (Admin)
  getStudents: (params?: { school_id?: string; class_id?: string; search?: string }) =>
    api.get('/api/admin/students', { params }),
    
  getStudent: (studentId: string) =>
    api.get(`/api/admin/students/${studentId}`),
    
  createStudent: (data: any) =>
    api.post('/api/admin/students', data),
    
  updateStudent: (studentId: string, data: any) =>
    api.put(`/api/admin/students/${studentId}`, data),
    
  deleteStudent: (studentId: string) =>
    api.delete(`/api/admin/students/${studentId}`),
  
  // Statistics
  getStats: () =>
    api.get('/api/admin/stats'),
    
  getClassStats: (classId: string) =>
    api.get(`/api/teachers/classes/${classId}/stats`),
    
  getCourseStats: (courseId: string) =>
    api.get(`/api/teachers/courses/${courseId}/stats`),
}