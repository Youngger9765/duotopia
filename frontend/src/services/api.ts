import axios from 'axios';

const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000';

const api = axios.create({
  baseURL: API_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

// Request interceptor to add auth token
api.interceptors.request.use((config) => {
  // Check for teacher token
  const teacherAuth = localStorage.getItem('auth-storage');
  if (teacherAuth) {
    const { state } = JSON.parse(teacherAuth);
    if (state?.token) {
      config.headers.Authorization = `Bearer ${state.token}`;
    }
  }

  // Check for student token
  const studentAuth = localStorage.getItem('student-auth-storage');
  if (studentAuth) {
    const { state } = JSON.parse(studentAuth);
    if (state?.token) {
      config.headers.Authorization = `Bearer ${state.token}`;
    }
  }

  return config;
});

// Response interceptor to handle errors
api.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response?.status === 401) {
      // Unauthorized - clear auth and redirect to login
      localStorage.removeItem('auth-storage');
      localStorage.removeItem('student-auth-storage');
      window.location.href = '/';
    }
    return Promise.reject(error);
  }
);

export default api;
