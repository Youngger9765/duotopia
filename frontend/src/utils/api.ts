import axios from 'axios';

export const apiClient = axios.create({
  baseURL: import.meta.env.VITE_API_URL || 'http://localhost:8080',
  headers: {
    'Content-Type': 'application/json',
  },
});

// 添加請求攔截器，自動帶上 token
apiClient.interceptors.request.use((config) => {
  const token = localStorage.getItem('token');
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});

// 添加響應攔截器，處理 401 錯誤
apiClient.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response?.status === 401) {
      // Token 過期或無效，清除並跳轉到登入頁
      localStorage.removeItem('token');
      window.location.href = '/';
    }
    return Promise.reject(error);
  }
);