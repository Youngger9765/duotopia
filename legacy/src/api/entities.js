import { base44 } from './base44Client';


export const ReadingAssessment = base44.entities.ReadingAssessment;

export const PracticeSession = base44.entities.PracticeSession;

export const Course = base44.entities.Course;

export const Lesson = base44.entities.Lesson;

export const StudentProgress = base44.entities.StudentProgress;

export const StudentImport = base44.entities.StudentImport;

export const TeacherNotification = base44.entities.TeacherNotification;

export const Class = base44.entities.Class;

export const ClassStudent = base44.entities.ClassStudent;

export const StudentTransfer = base44.entities.StudentTransfer;

export const ClassCourseMapping = base44.entities.ClassCourseMapping;

export const School = base44.entities.School;

export const SchoolTeacher = base44.entities.SchoolTeacher;

export const SchoolStudent = base44.entities.SchoolStudent;

export const Semester = base44.entities.Semester;

export const ClassTeacher = base44.entities.ClassTeacher;

export const ActivityResult = base44.entities.ActivityResult;

export const SpeakingQuiz = base44.entities.SpeakingQuiz;

export const SpeakingQuizResult = base44.entities.SpeakingQuizResult;

export const SpeakingSession = base44.entities.SpeakingSession;

export const StudentSession = base44.entities.StudentSession;

export const StudentAssignment = base44.entities.StudentAssignment;

export const RecentTeacher = base44.entities.RecentTeacher;

export const AssignmentSubmission = base44.entities.AssignmentSubmission;

export const SpeakingScenarioResult = base44.entities.SpeakingScenarioResult;



// auth sdk:
export const User = base44.auth;