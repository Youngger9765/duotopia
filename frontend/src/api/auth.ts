import type { User } from '@/types'
import { api } from '@/lib/api'

export const authApi = {
  // 主要登入方法，包含角色判斷邏輯
  login: async (email: string, password: string) => {
    // This is already implemented correctly in LoginPage.tsx
    // Just need to parse the response properly
    const formData = new URLSearchParams()
    formData.append('username', email)
    formData.append('password', password)
    
    const response = await api.post('/api/auth/login', formData, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      }
    })
    
    const data = response.data
    
    // Determine roles based on user type
    let roles = []
    if (data.is_individual_teacher) {
      roles.push('individual_teacher')
    }
    if (data.is_institutional_admin) {
      roles.push('institutional_admin')
    }
    if (data.user_type === 'admin') {
      roles.push('admin')
    }
    if (roles.length === 0) {
      roles.push('teacher') // Default role
    }
    
    // Return the response in the expected format
    return {
      access_token: data.access_token,
      user: {
        id: data.user_id,
        email: email,
        full_name: data.full_name || email,
        name: data.full_name || email,
        role: roles[0], // Primary role
        has_multiple_roles: roles.length > 1,
        is_individual_teacher: data.is_individual_teacher || false
      } as User,
      roles: roles,
    }
  },

  validateToken: async (token: string) => {
    // Call the backend API to validate token and get user info
    try {
      const response = await api.get('/api/auth/validate', {
        headers: {
          'Authorization': `Bearer ${token}`
        }
      })
      
      return {
        id: response.data.user_id,
        email: response.data.email,
        full_name: response.data.full_name,
        name: response.data.full_name
      } as User
    } catch (error) {
      // If validation fails, throw error to force re-login
      throw new Error('Token validation failed')
    }
  },

  // Google OAuth 登入
  googleLogin: async (idToken: string) => {
    const response = await api.post('/api/auth/google', { id_token: idToken })
    return response.data
  },

  // 學生登入
  studentLogin: async (email: string, birthDate: string) => {
    const response = await api.post('/api/auth/student/login', { 
      email, 
      birth_date: birthDate 
    })
    return response.data
  },

  // 註冊新用戶
  register: async (data: any) => {
    const response = await api.post('/api/auth/register', data)
    return response.data
  },

  // 驗證 token
  validate: async () => {
    const response = await api.get('/api/auth/validate')
    return response.data
  }
}