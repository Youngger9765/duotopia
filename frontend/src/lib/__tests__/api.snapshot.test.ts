/* eslint-disable @typescript-eslint/no-explicit-any */
/**
 * API Snapshot Testing
 * 確保 API 回應格式不會意外改變
 */
import { describe, it, expect, vi } from "vitest";

// Mock the api module
vi.mock("../api", () => ({
  apiClient: {
    teacherLogin: vi.fn(),
    getTeacherDashboard: vi.fn(),
    getSubmission: vi.fn(),
  },
}));

import { apiClient } from "../api";

describe("API Response Snapshots", () => {
  it("should match login response snapshot", async () => {
    const mockResponse = {
      access_token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
      token_type: "bearer",
      user_type: "teacher",
      user: {
        id: 1,
        email: "teacher@test.com",
        name: "Test Teacher",
        is_demo: false,
      },
    };

    vi.mocked(apiClient.teacherLogin).mockResolvedValue(mockResponse);

    const result = await apiClient.teacherLogin({
      email: "teacher@test.com",
      password: "password123",
    });

    expect(result).toMatchSnapshot();
  });

  it("should match dashboard response snapshot", async () => {
    const mockDashboard = {
      teacher: {
        id: 1,
        name: "Test Teacher",
        email: "teacher@test.com",
      },
      classroom_count: 3,
      student_count: 25,
      recent_assignments: [
        {
          id: 31,
          title: "English Essay",
          due_date: "2025-09-20T23:59:59Z",
          submission_count: 12,
          total_students: 15,
        },
      ],
      statistics: {
        assignments_created: 45,
        total_submissions: 234,
        avg_score: 85.6,
      },
    };

    vi.mocked(apiClient.getTeacherDashboard).mockResolvedValue(mockDashboard);

    const result = await apiClient.getTeacherDashboard();

    expect(result).toMatchSnapshot();
  });

  it("should match submission response snapshot", async () => {
    const mockSubmission = {
      id: 1,
      assignment_id: 31,
      student_id: 1,
      student: {
        id: 1,
        student_id: "S001",
        name: "Test Student",
        email: "student@test.com",
      },
      content: {
        text_response: "This is my essay...",
        attachments: [],
        audio_recording: null,
      },
      status: "submitted",
      score: null,
      feedback: null,
      submitted_at: "2025-09-15T10:30:00Z",
      graded_at: null,
    };

    vi.mocked(apiClient.getSubmission).mockResolvedValue(mockSubmission);

    const result = await apiClient.getSubmission(31, 1);

    expect(result).toMatchSnapshot();
  });

  it("should match error response snapshot", async () => {
    const mockError = new Error("Could not validate credentials");
    mockError.name = "ApiError";
    // 模擬 API 錯誤結構
    (mockError as any).status = 401;
    (mockError as any).detail = "Could not validate credentials";

    vi.mocked(apiClient.getSubmission).mockRejectedValue(mockError);

    try {
      await apiClient.getSubmission(31, 1);
    } catch (error: any) {
      expect({
        message: error.message,
        status: error.status,
        detail: error.detail,
      }).toMatchSnapshot();
    }
  });
});
