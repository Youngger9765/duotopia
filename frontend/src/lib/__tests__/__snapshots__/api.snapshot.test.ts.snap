// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API Response Snapshots > should match dashboard response snapshot 1`] = `
{
  "classroom_count": 3,
  "recent_assignments": [
    {
      "due_date": "2025-09-20T23:59:59Z",
      "id": 31,
      "submission_count": 12,
      "title": "English Essay",
      "total_students": 15,
    },
  ],
  "statistics": {
    "assignments_created": 45,
    "avg_score": 85.6,
    "total_submissions": 234,
  },
  "student_count": 25,
  "teacher": {
    "email": "teacher@test.com",
    "id": 1,
    "name": "Test Teacher",
  },
}
`;

exports[`API Response Snapshots > should match error response snapshot 1`] = `
{
  "detail": "Could not validate credentials",
  "message": "Could not validate credentials",
  "status": 401,
}
`;

exports[`API Response Snapshots > should match login response snapshot 1`] = `
{
  "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
  "token_type": "bearer",
  "user": {
    "email": "teacher@test.com",
    "id": 1,
    "is_demo": false,
    "name": "Test Teacher",
  },
  "user_type": "teacher",
}
`;

exports[`API Response Snapshots > should match submission response snapshot 1`] = `
{
  "assignment_id": 31,
  "content": {
    "attachments": [],
    "audio_recording": null,
    "text_response": "This is my essay...",
  },
  "feedback": null,
  "graded_at": null,
  "id": 1,
  "score": null,
  "status": "submitted",
  "student": {
    "email": "student@test.com",
    "id": 1,
    "name": "Test Student",
    "student_id": "S001",
  },
  "student_id": 1,
  "submitted_at": "2025-09-15T10:30:00Z",
}
`;
